runtime/io-error.cpp

```diff
@@ -76,6 +77,32 @@ bool IoErrorHandler::GetIoMsg(char *buffer, std::size_t bufferLength) {
     ToFortranDefaultCharacter(buffer, bufferLength, msg);
     return true;
   }
+
+  char *newBuf;
+  // Following code is taken from llvm/lib/Support/Errno.cpp
+  // in LLVM v9.0.1
+#if HAVE_STRERROR_R
+  // strerror_r is thread-safe.
+#if defined(__GLIBC__) && defined(_GNU_SOURCE)
+  // glibc defines its own incompatible version of strerror_r
+  // which may not use the buffer supplied.
+  newBuf = ::strerror_r(ioStat_, buffer, bufferLength);
+#else
   return ::strerror_r(ioStat_, buffer, bufferLength) == 0;
+#endif
+#elif HAVE_DECL_STRERROR_S // "Windows Secure API"
+  return ::strerror_s(buffer, bufferLength, ioStat_) == 0;
+#elif HAVE_STRERROR
+  // Copy the thread un-safe result of strerror into
+  // the buffer as fast as possible to minimize impact
+  // of collision of strerror in multiple threads.
+  newBuf = strerror(ioStat_);
+#else
+  // Strange that this system doesn't even have strerror
+  return false;
```

Comment:
Perhaps #error on systems without strerror would help people who are porting this code in the future?

Reply:
I've added an error in CMake code